File: Leetcode/9 - Palindrome Number/solution.py
def isPalindrome(self, x: int) -> bool:
    if x < 0 or ((x % 10 == 0) and (x != 0)):
        return False
    reverse: int = 0
    while (x > reverse):
        reverse = (reverse * 10) + (x % 10)
        x //= 10
    return (x == reverse) or (x == reverse // 10)

File: Leetcode/9 - Palindrome Number/solution.ts
function isPalindrome(x: number): boolean {
    if (x < 0 || ((x % 10 === 0) && (x !== 0))) return false;
    let reverse: number = 0;
    while (x > reverse) {
        reverse = (reverse * 10) + (x % 10);
        x = Math.trunc(x / 10);
    }
    return (x === reverse) || (x === Math.trunc(reverse / 10));
};